### PR TITLE
Add --use-modular-headers flag to `push`

### PR DESCRIPTION
**Pull request by amorde**
Closes https://github.com/CocoaPods/CocoaPods/issues/7683

Continuation of https://github.com/CocoaPods/CocoaPods/pull/7959

### CODE REVIEW

File: spec/command/trunk/push_spec.rb [hunk 1]
```diff
@@ -183,6 +183,13 @@ def found_podspec_among_files(files)
         cmd = Command.parse(%w(trunk push spec/fixtures/BananaLib.podspec --skip-tests))
         cmd.send(:validate_podspec)
       end
+
+      it 'passes use modular headers' do
```
Comment (amorde) [inline spec/command/trunk/push_spec.rb]: I wanted to make a test that matched https://github.com/CocoaPods/CocoaPods/blob/89a273882d75f42e51e486410dbab4fc15292a09/spec/functional/command/repo/push_spec.rb#L224-L236, but I had some issues trying to update the version of CocoaPods used with Bundler

File: lib/pod/command/trunk/push.rb [hunk 2]
```diff
@@ -119,7 +121,10 @@ def validate_podspec
           validator = Validator.new(spec, %w(https://github.com/CocoaPods/Specs.git))
           validator.allow_warnings = @allow_warnings
           validator.use_frameworks = @use_frameworks
-          validator.swift_version  = @swift_version if Validator.method_defined?(:swift_version=)
+          if Validator.method_defined?(:use_modular_headers=)
```
Comment (segiddins) [inline lib/pod/command/trunk/push.rb]: `if validator.respond_to?(:use_modular_headers=)`

Comment (dnkoutso) [reply lib/pod/command/trunk/push.rb]: could also change the above too?

Comment (amorde) [reply lib/pod/command/trunk/push.rb]: 👍

Comment (amorde) [reply lib/pod/command/trunk/push.rb]: also the reason I used a separate `if` check instead of inline was for rubocop line length

**Pull request merged**